File: Algorithms/Stack/20. Valid Parentheses.py
class Solution(object):
	def isValid(self, s):
		stack = list()
		Mapping = {'}': '{', ']': '[', ')': '('}
		leftBracket = ['{', '[', '(']
		for char in s:
			if(char in leftBracket):
				stack.append(char)
			elif(char in Mapping):
				if(not stack or stack.pop() != Mapping[char]):
					return False
		return not stack

File: Algorithms/String/20. Valid Parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
		unordered_map<char, char> mapping({{'}', '{'}, {')', '('}, {']', '['}});
		unordered_set<char> leftBracket({'(', '[', '{'});
		vector<char> stack;
		for(char& c : s)
		{
			if(leftBracket.count(c) != 0)
			{
				stack.emplace_back(c);
			}
			else
			{
				if(stack.empty() || stack.back() != mapping[c]) return false;
				stack.pop_back();
			}
		}
		return stack.empty();
    }
};

